homeworks/src/main/java/hw31/jdbc/OrderRepository.java

```diff
@@ -0,0 +1,133 @@
+        } catch (SQLException e) {
+            throw new JdbcOperationException("Can't extract order from DB by ID %d".formatted(id));
+        }
+    }
+
+    public List<Order> getAllOrders() {
+        List<Order> orders = new ArrayList<>();
+        String sql = SQLQuery.GET_ALL_ORDERS;
+        try (Connection connection = DBConnection.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(sql);
+             ResultSet resultSet = preparedStatement.executeQuery()) {
+            while (resultSet.next()) {
+                Order order = mapOrder(resultSet);
+                orders.add(order);
+            }
+            return orders;
+        } catch (SQLException e) {
+            throw new JdbcOperationException("Can't extract orders from DB");
+        }
+    }
+
+    public void addOrder(Product product) {
+        Objects.requireNonNull(product);
+        String sql = SQLQuery.ADD_ORDER;
+        try (Connection connection = DBConnection.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setDouble(1, product.getCost());
+            preparedStatement.setInt(2, product.getId());
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Order order = mapOrder(resultSet);
+                saveOrderProducts(order, product);
+            }
+        } catch (SQLException e) {
+            throw new JdbcOperationException("Can't add order with product %s".formatted(product));
+        }
+    }
+
+    private void saveOrderProducts(Order order, Product product) {
+        Objects.requireNonNull(order);
+        Objects.requireNonNull(product);
+        productRepository.addProduct(product);
+        linkOrderToProduct(order.getId(), product.getId());
+    }
+
+    private void linkOrderToProduct(int orderId, int productId) {
+        if (orderId < 1 || productId < 1) {
+            throw new JdbcOperationException("Illegal arguments: Product id is " + productId + ", order id is " + orderId);
+        }
+        String sql = SQLQuery.LINK_PRODUCT_TO_ORDER;
+        try (Connection connection = DBConnection.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setInt(1, orderId);
+            preparedStatement.setInt(2, productId);
+            int result = preparedStatement.executeUpdate();
+            if (result < 1) {
+                throw new JdbcOperationException("Product " + productId + " was not added to order " + orderId);
+            }
+        } catch (SQLException e) {
+            throw new JdbcOperationException("Can't link product " + productId + " to order " + orderId);
+        }
+    }
+
+    private Order mapOrder(ResultSet resultSet) {
+        Objects.requireNonNull(resultSet);
+        try {
+            int id = resultSet.getInt(1);
```

Comment:
Indexes work faster, but sometimes you would need to extract a column from the result set by its name. Just keep in mind, that JDBC API provides you with this functionality.